plugins/nf-nomad/src/main/nextflow/nomad/NomadConfig.groovy

```diff
@@ -66,18 +78,23 @@ class NomadConfig {
         final AffinitySpec affinitySpec
         final ConstraintSpec constraintSpec
 
-        NomadJobOpts(Map nomadJobOpts){
+        NomadJobOpts(Map nomadJobOpts, Map<String,String> env=null){
+
+            sysEnv = env==null ? new HashMap<String,String>(System.getenv()) : env
+
             deleteOnCompletion = nomadJobOpts.containsKey("deleteOnCompletion") ?
                     nomadJobOpts.deleteOnCompletion : false
             if( nomadJobOpts.containsKey("datacenters") ) {
                 datacenters = ((nomadJobOpts.datacenters instanceof List<String> ?
                         nomadJobOpts.datacenters : nomadJobOpts.datacenters.toString().split(","))
                         as List<String>).findAll{it.size()}.unique()
             }else{
-                datacenters = []
+                datacenters = List.of(sysEnv.get('NOMAD_DC'))
             }
-            region = nomadJobOpts.region ?: null
-            namespace = nomadJobOpts.namespace ?: null
+
+            region = nomadJobOpts.region ?: sysEnv.get('NOMAD_REGION')
```

Comment:
I think we want the opposite: System env has priority over nomadJobOpts

Reply:
Actually Jorge, the general design is if something is specified within the Nextflow layer then use it otherwise, try to source it from the environment 

For example 

https://github.com/nextflow-io/nextflow/blob/0a8a484e082d0be4a41eec9c4ae9c298f9822f19/plugins/nf-azure/src/main/nextflow/cloud/azure/config/AzBatchOpts.groovy#L64

Reply:
ummm I think it's an "error"

General idea in microservices frameworks, for example, is to use System Env as high priority as this is an easy way to overwrite a "fixed" configuration and tests different configuration without change any line

Imagine following scenario:
- I have a pipeline with a `EU` region in the config
- I would like to run the pipeline against a new region to verify is working. Running it with `NOMAD_REGION=US nextflow run mypipeline` will work without change anything

To achieve the same result we can run the pipeline overwritting the config value with a param (i.e. --region=US) but IMHO is not so flexible

Reply:
Mm, okay so I do see the reasoning behind this logic and it does make sense in our use-case. 

However, considering that our users would already be people who're used to Nextflow config management, they would not ideally specific those fields in the config itself (such as `nomad.client.address` ) and purely specify it via the env vars which they can switch easily using build config/shell scripts etc.


Let's track this as a discussion so that we know this is a possible enhancement we can have in future - what do you think?

If there's an interest in a different priority order, then we have a good reason for using ENV VARS as a priority.

Reply:
Done, here's the poll https://github.com/nextflow-io/nf-nomad/discussions/48 :)